mongo/options/clientoptions.go

```diff
@@ -300,187 +322,305 @@ func (c *ClientOptions) ApplyURI(uri string) *ClientOptions {
+// values:
+//
+// 1. "connect=direct" for direct connections
+//
+// 2. "connect=automatic" for automatic discovery.
+//
+// The default is false.
 func (c *ClientOptions) SetDirect(b bool) *ClientOptions {
 	c.Direct = &b
 	return c
 }
 
-// SetHeartbeatInterval specifies the interval to wait between server monitoring checks.
+// SetHeartbeatInterval specifies the amount of time to wait between periodic background server checks. This can also be
+// set through the "heartbeatIntervalMS" URI option (e.g. "heartbeatIntervalMS=10000"). The default is 10 seconds.
 func (c *ClientOptions) SetHeartbeatInterval(d time.Duration) *ClientOptions {
 	c.HeartbeatInterval = &d
 	return c
 }
 
-// SetHosts specifies the initial list of addresses from which to discover the rest of the cluster.
+// SetHosts specifies a list of host names or IP addresses for servers in a cluster. Both IPv4 and IPv6 addresses are
+// supported. IPv6 literals must be enclosed in '[]' following RFC2732 syntax.
+//
+// Hosts can also be specified as a comma-separated list in a URI. For example, to include "localhost:27017" and
+// "localhost:27018", a URI could be "mongodb://localhost:27017,localhost:27018". The default is ["localhost:27017"]
 func (c *ClientOptions) SetHosts(s []string) *ClientOptions {
 	c.Hosts = s
 	return c
 }
 
-// SetLocalThreshold specifies how far to distribute queries, beyond the server with the fastest
-// round-trip time. If a server's roundtrip time is more than LocalThreshold slower than the
-// the fastest, the driver will not send queries to that server.
+// SetLocalThreshold specifies the width of the 'latency window': when choosing between multiple suitable servers for an
+// operation, this is the acceptable non-negative delta between shortest and longest average round-trip times. Servers
+// within the latency. This can also be set through the "localThresholdMS" URI option (e.g. "localThresholdMS=15000").
+// The default is 15 milliseconds.
 func (c *ClientOptions) SetLocalThreshold(d time.Duration) *ClientOptions {
 	c.LocalThreshold = &d
 	return c
 }
 
-// SetMaxConnIdleTime specifies the maximum number of milliseconds that a connection can remain idle
-// in a connection pool before being removed and closed.
+// SetMaxConnIdleTime specifies the maximum amount of time that a connection can remain in a connection pool without
+// being used before it is removed and closed. This can also be set through the "maxIdleTimeMS" URI option (e.g.
+// "maxIdleTimeMS=10000"). The default is 0, meaning a connection can remain unused indefinitely.
 func (c *ClientOptions) SetMaxConnIdleTime(d time.Duration) *ClientOptions {
 	c.MaxConnIdleTime = &d
 	return c
 }
 
-// SetMaxPoolSize specifies the max size of a server's connection pool.
+// SetMaxPoolSize specifies the maximum number of connections allowed in each server's connection pool. Requests to a
```

Comment:
"allowed in each server's connection pool"

This may mislead people into thinking it's controlling the connection pool in the MongoDB server rather than the driver's connection pool _to_ that server.  

Try rewording.

Reply:
Good point. Changed to "SetMaxPoolSize specifies that maximum number of connections allowed in the driver's connection pool to each server.".